src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -49,50 +41,84 @@ TileProvider.prototype.preprocessDataLayer = function preprocessLayer(layer, vie
     });
 };
 
+function defer() {
+    const deferedPromise = {};
+    deferedPromise.promise = new Promise((resolve, reject) => {
+        deferedPromise.resolve = resolve;
+        deferedPromise.reject = reject;
+    });
+    return deferedPromise;
+}
+
 TileProvider.prototype.executeCommand = function executeCommand(command) {
-    var extent = command.extent;
+    const extent = command.extent;
     if (command.requester &&
         !command.requester.material) {
         // request has been deleted
         return Promise.reject(new CancelledCommandException(command));
     }
 
-    var parent = command.requester;
-
-
-    // build tile
-    var params = {
-        layerId: command.layer.id,
-        extent,
-        level: (command.level === undefined) ? (parent.level + 1) : command.level,
-        segment: 16,
-        materialOptions: command.layer.materialOptions,
-        disableSkirt: command.layer.disableSkirt,
-    };
+    const parent = command.requester;
+    const level = (command.level === undefined) ? (parent.level + 1) : command.level;
 
-    const geometry = new TileGeometry(params, command.layer.builder);
+    if (!this.cacheGeometry[level]) {
+        this.cacheGeometry[level] = new Map();
+    }
 
-    var tile = new TileMesh(geometry, params);
+    const ce = command.layer.getCommonGeometryExtent(extent);
+    const south = ce.south().toFixed(8);
+
+    // build geometry if doesn't exist
+    if (!this.cacheGeometry[level].get(south)) {
```

Comment:
This is wrong: the `south` value alone can't serve as a cache key.
You must at least depend on `command.layer.disableSkirt` (and probably on ellipsoidSize - even if currently this cannot be changed) and the `builder` used.

Reply:
yes, in another PR, there's not user cases for the moment